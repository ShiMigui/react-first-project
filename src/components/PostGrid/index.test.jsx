import { render, screen } from '@testing-library/react';
import { PostGrid } from '.';

const posts = [
    { title: 'postcard 1', body: 'hello 1', url: 'generic.png' },
    { title: 'postcard 2', body: 'hello 2', url: 'generic.png' },
    { title: 'postcard 3', body: 'hello 3', url: 'generic.png' },
]

const grid = <PostGrid posts={posts} />;

describe('<PostGrid />', () => {
    it('must contains 3 postcards', () => {
        render(grid);

        let arr = screen.getAllByRole('heading', { name: /postcard/i });
        arr.map(p => expect(p).toBeInTheDocument());
        expect(arr).toHaveLength(3);

        expect(screen.getAllByRole('img', { name: /postcard/i })).toHaveLength(3);

        expect(screen.getAllByText(/hello/i)).toHaveLength(3);
    })

    it('must match with snapshot', () => {
        const { container } = render(grid);

        expect(container.firstChild).toMatchSnapshot();
    })
});