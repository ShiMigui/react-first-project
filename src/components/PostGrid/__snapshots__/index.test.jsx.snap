// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostGrid /> must match with snapshot 1`] = `
<div
  class="post-grid"
>
  <div
    class="post-card"
    data-testid="post-card"
  >
    <img
      alt="postcard 1"
      src="generic.png"
    />
    <div
      class="post-content"
    >
      <h4>
        postcard 1
      </h4>
      <p>
        hello 1
      </p>
    </div>
  </div>
  <div
    class="post-card"
    data-testid="post-card"
  >
    <img
      alt="postcard 2"
      src="generic.png"
    />
    <div
      class="post-content"
    >
      <h4>
        postcard 2
      </h4>
      <p>
        hello 2
      </p>
    </div>
  </div>
  <div
    class="post-card"
    data-testid="post-card"
  >
    <img
      alt="postcard 3"
      src="generic.png"
    />
    <div
      class="post-content"
    >
      <h4>
        postcard 3
      </h4>
      <p>
        hello 3
      </p>
    </div>
  </div>
</div>
`;
